hathor/websocket/streamer.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2024 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import deque
+from typing import TYPE_CHECKING, Any, AsyncIterator, Iterable
+
+from twisted.internet.defer import Deferred
+from twisted.internet.interfaces import IPushProducer
+from zope.interface import implementer
+
+from hathor.manager import HathorManager
+from hathor.types import AddressB58
+from hathor.util import json_dumpb
+
+if TYPE_CHECKING:
+    from hathor.websocket.protocol import HathorAdminWebsocketProtocol
+
+
+class ManualAddressSequencer:
+    def __init__(self, addresses: Iterable[AddressB58]) -> None:
+        self.pending_addresses: deque[AddressB58] = deque(addresses)
+        self.stop = False
+        self.await_items: Deferred | None = None
+
+    def __aiter__(self) -> AsyncIterator[tuple[int, AddressB58]]:
+        return self._async_iter()
+
+    def _resume_iter(self) -> None:
+        if self.await_items is None:
+            return
+        self.await_items.callback(None)
+
+    def add_addresses(self, addresses: list[AddressB58], last: bool) -> None:
+        self.pending_addresses.extend(addresses)
+        if last:
+            self.stop = True
+        self._resume_iter()
+
+    async def _async_iter(self) -> tuple[int, AddressB58]:
+        idx = 0
+        while True:
+            while self.pending_addresses:
+                address = self.pending_addresses.popleft()
+                yield (idx, address)
```

Comment:
Validate whether `address` is a valid base58 address.

Reply:
Done!

Reply:
It's kind of enough since the index would raise an exception if the size is wrong. Now that we have this validation, the only consequence of an invalid address is to report zero transactions. We can add more basic checks like the first letter and the checksum but I guess it's not worth.

Reply:
Ok!